File: Q16 Subarray sum equals to k.py
class Solution:
    def subarraySum(self, nums, k):
        n = len(nums)
        count = 0
        sum_val = 0
        mp = {0: 1}
        
        for i in range(n):
            sum_val += nums[i]
            remove = sum_val - k
            if remove in mp:
                count += mp[remove]
            if sum_val in mp:
                mp[sum_val] += 1
            else:
                mp[sum_val] = 1
        
        return count

File: Q16 subarray sum equals to k.js
var subarraySum = function(nums, k) {
    let n = nums.length;
    let count = 0;
    let sum = 0;
    let mp = new Map();
    mp.set(0, 1);
    
    for (let i = 0; i < n; i++) {
        sum += nums[i];
        let remove = sum - k;
        if (mp.has(remove)) {
            count += mp.get(remove);
        }
        if (mp.has(sum)) {
            mp.set(sum, mp.get(sum) + 1);
        } else {
            mp.set(sum, 1);
        }
    }
    
    return count;
};
